plugins/cuda/cuda_plugin.c

```diff
@@ -470,6 +470,22 @@ int cuda_plugin_resume_devices_late(int pid)
 }
 CR_PLUGIN_REGISTER_HOOK(CR_PLUGIN_HOOK__RESUME_DEVICES_LATE, cuda_plugin_resume_devices_late)
 
+static bool has_gpu()
+{
+	DIR *dir = opendir("/proc/driver/nvidia/gpus/");
```

Comment:
nit: I think it is enough to check an existence of  /proc/driver/nvidia/gpus. I don't think that someone loads a module without having any nvidia card on the host.